usr/sbin/pkcsslotd/garbage_linux.c

```diff
@@ -80,8 +81,8 @@ BOOL StartGCThread(Slot_Mgr_Shr_t *MemPtr)
 #ifdef DEV
     // Only development builds
     LogLog("StartGCThread: garbage collection thread started as ID "
-           "%d (%#x) by ID %d (%#x)",
-           GCThread, GCThread, pthread_self(), pthread_self());
+           "%lu by ID %lu",
+           GCThread, pthread_self());
```

Comment:
Is it save to assume that it will be the same on all Linuxes? I don't think OCK will compile on non-Linux platforms anyway. So I would not spend much effort to make it portable to non-Linux platforms as of now.

Reply:
id bet on an unsigned int type on linux platforms. you could just use a wrapper around thread id related stuff that treads it as on opaque uchar array of sizeof(pthread_t) length and convert it to an unsigned long according to host byte order. dont know how much work that would require

Reply:
Those lines are included only if compiled with -DDEV to enable development mode build of pkcsslotd (which is not the default). One must explicitly specify -DDEV at configure time to enable that. 
So I would not want to spend much effort on this at the moment.